### PR TITLE
Add Image Slide Viewer & Image Viewer Header

### PR DESCRIPTION
**Pull request by samsep**
ping @vietnogi @aselbie

### CODE REVIEW

File: example/scripts/image-viewer-example.controller.coffee [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+ImageViewerExampleController = ($scope, $stateParams) ->
+  vm       = this
+  vm.userHandle = "batman"
+  vm.userAvatar = "http://www.topcoder.com/i/m/cardiboy_big.jpg"
+  vm.showNotifications = false
+  vm.files = [
+    {
+      fileId: "abc",
+      path: "work/1447877190406-e9c02ed7-00bf-4e52-8c57-4d73fec009ac/9be085da-2013-44e9-8701-507a03c6716e/Screen Shot 2015-12-01 at 1.13.17 PM.png",
```
Comment (vietnogi) [inline example/scripts/image-viewer-example.controller.coffee]: put files into an example/images and use that instead

File: example/scripts/image-viewer-example.controller.coffee [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      caption: "this is an image",
+      url: "https://media.licdn.com/mpr/mpr/p/4/005/05b/0ad/17ef3d8.jpg"
+    },
+    {
+      fileId: "def",
```
Comment (vietnogi) [inline example/scripts/image-viewer-example.controller.coffee]: Do we need `fileId`?

File: src/scripts/controllers/image-viewer-header.controller.coffee [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  vm = this
+  vm.avatar = $scope.avatar
+  vm.handle = $scope.handle
+  vm.title = $scope.title
+  vm.commentsAllowed = $scope.commentsAllowed
+  vm.downloadAllowed = $scope.downloadAllowed
+  vm.toggleComments = $scope.toggleComments
+
+  vm.generateProfileUrl = (handle) ->
+    "https://www.topcoder.com/members/#{handle}"
```
Comment (aselbie) [inline src/scripts/controllers/image-viewer-header.controller.coffee]: Should this use "API_URL" constant, or do we only every hit topcoder.com for the profiles?

File: src/views/image-viewer-header.directive.jade [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      a(href="{{vm.generateProfileUrl(vm.handle)}}" target="_blank")
+        avatar(avatar-url="{{ vm.avatar }}")
+
+      a(href="{{vm.generateProfileUrl(vm.handle)}}" target="_blank")
+        p.name {{ vm.handle }}
+
+
+    .icons
+      button.clean(ng-if="vm.downloadAllowed")
+        a(href="{{ vm.file.url }}" target="_blank")
```
Comment (aselbie) [inline src/views/image-viewer-header.directive.jade]: I don't think we have a file in this case. Maybe pass another property through the directive for "downloadUrl"?

File: src/scripts/controllers/image-viewer-header.controller.coffee (hunk 3 above)
Comment (vietnogi) [inline src/scripts/controllers/image-viewer-header.controller.coffee]: also, is this for the avatar?  if so, the logic should be in the avatar element

Comment (samsep) [inline src/scripts/controllers/image-viewer-header.controller.coffee]: @aselbie  the latter, it actually opens their profile page in a new tab. @vietnogi keeping the logic on the controller for now to keep avatar more versatile (not every avatar would have this functionality)

**Comment by vietnogi** [comment]
couldn't the avatar have an option to toggle if its a link or not?

**Comment by samsep** [comment]
yeah will refactor this later

**Pull request merged**